### PR TITLE
[merged] pull: Support multiple specifications of --subpath

### PR DESCRIPTION
**Pull request by alexlarsson**
I need this in flatpak to avoid doing multiple pulls when doing
locale subsetting.

### CODE REVIEW

**Comment by cgwalters** [comment]
Overall looks sane, there is just some nontrivial code here I want to take a bit of time later to read through carefully.

**Comment by alexlarsson** [comment]
I pushed a (trivially) rebased version that applies again.

File: src/libostree/ostree-repo-pull.c [hunk 1]
```diff
@@ -1101,10 +1157,10 @@ scan_commit_object (OtPullData         *pull_data,
         goto out;
 
       queue_scan_one_metadata_object_c (pull_data, tree_contents_csum_bytes,
-                                        OSTREE_OBJECT_TYPE_DIR_TREE, recursion_depth + 1);
+                                        OSTREE_OBJECT_TYPE_DIR_TREE, g_strdup ("/"), recursion_depth + 1);
```
Comment (cgwalters) [inline src/libostree/ostree-repo-pull.c]: This is a leak, right?

File: src/libostree/ostree-repo-pull.c [hunk 2]
```diff
@@ -381,12 +388,80 @@ static void
 enqueue_one_object_request (OtPullData        *pull_data,
                             const char        *checksum,
                             OstreeObjectType   objtype,
+                            const char        *path,
                             gboolean           is_detached_meta,
                             gboolean           object_is_stored);
 
 static gboolean
+matches_pull_dir (const char *current_file,
```
Comment (cgwalters) [inline src/libostree/ostree-repo-pull.c]: I had to read this twice at least, but it does make sense.

Comment (alexlarsson) [inline src/libostree/ostree-repo-pull.c]: If it was hard to write it should be hard to read...

Honestly, I don't know why things like this are so tricky. I wonder if there is a way to formulate it that is better.

**Comment by cgwalters** [comment]
@rh-atomic-bot r+ 4084be1

**Pull request closed without merging**